packages/search-common/api-report.md

```diff
@@ -27,11 +29,21 @@ export interface DocumentDecorator {
 //
 // @public
 export interface IndexableDocument {
+  authorization?: {
+    resourceRef: string;
```

Comment:
Would it make sense to move `visibilityPermission` here? Perhaps as just `permission`. Feels it pulls together the data model better as it currently seems a bit scattered. Would also allow for more flexibility in the collator implementations right? Being able to return documents with varying permissions.

Might be missing something though, I didn't dive too deep 😅

Reply:
It actually started out that way - some commentary on why it ended up like this in the first bullet of the PR summary. Happy to continue discussing!

> dea9f08 and 058df41 together add the information needed to authorize search results. The former adds a field to the DocumentCollator interface to hold the permission, and the latter adds the resourceRef to the documents. By organizing the information in this way we encode the constraint that all documents in a given index must be authorized using the same permission. This is not actually required for result-by-result authorization, but if we implement in-engine authorization in the future the constraint will likely make the work a lot simpler.

Reply:
Aight, makes sense 👍